docs/cloud/pouta/tutorials/image-pipeline.md

```diff
@@ -0,0 +1,650 @@
+export OUTPUT_BUCKET="output_bucket"
+
+source /home/ubuntu/db_cred.sh
+    
+export NEGATED_PREFIX="negated_"
+
+(
+# perform the task if the lock is free, otherwise exit
+flock -n 200 || exit 1
+# iterate over the images in the bucket
+for IMAGE_URL in `s3cmd ls s3://$INPUT_BUCKET | awk '{ print $4 }'`
+do
+        # get the current timestamp
+        TIMESTAMP=$(date +%FT%T)
+        # get the image
+        IMAGE_NAME=$(echo "$IMAGE_URL" | awk -F '/' '{ print $NF }')
+        s3cmd get $IMAGE_URL $IMAGE_NAME
+        # compute the negated
+        NEGATED_IMAGE_NAME="$NEGATED_PREFIX$IMAGE_NAME"
+        convert -negate $IMAGE_NAME $NEGATED_IMAGE_NAME
+        # compute hash of the result
+        NEGATED_IMAGE_HASH=$(sha256sum $NEGATED_IMAGE_NAME | awk '{ print $1 }')
+        # log to the db that this was done
+        psql -h "$DB_HOST" -p "$DB_PORT" -U "$DB_USER" -d "$DB_NAME" -c "INSERT INTO log_records (timestamp, negated_image_name, negated_image_hash) VALUES ('$TIMESTAMP', '$NEGATED_IMAGE_NAME', '$NEGATED_IMAGE_HASH')"
+        # push the negated image to the output bucket
+        s3cmd put $NEGATED_IMAGE_NAME s3://$OUTPUT_BUCKET
+        # delete the local copies of the images
+        rm $IMAGE_NAME $NEGATED_IMAGE_NAME
+        # delete the image from the input bucket
+        s3cmd del $IMAGE_URL
+        # sleep 1 sec
+        sleep 1
+done
+) 200>/var/lock/pipeline_script_lock
+```
+As done earlier as well, we save the content of the file by pressing `CTRL + X`, then `y`, and finally `Enter` key.
+
+The script now contains the logic of the pipeline, but it is not set to run automatically yet.
+On `terminal_C` we run the following commands:
+```
+$ chmod +x /home/ubuntu/pipeline_script.sh
+$ crontab -l > crontab_list
+$ echo "* * * * * /home/ubuntu/pipeline_script.sh" >> crontab_list
+$ crontab crontab_list
+```
+
+Now the script is configured to run automatically every minute.
+We are ready to test the functioning of the pipeline.
+
+## Step 5: testing the pipeline
+
+We pick up an image of choice and we copy it to the home folder of our workstation.
+As an example, here is our image at `~/cat1.jpg`:
+
+![image-cat](../../img/cat1.jpg)
+
+We now want to send the image through the pipeline.
+On `terminal_A`, we first navigate to the home folder and then upload the image to Allas:
+```
+$ cd ~
+$ s3cmd put cat1.jpg s3://input_bucket
+upload: 'cat1.jpg' -> 's3://input_bucket/cat1.jpg'  [1 of 1]
+ 133275 of 133275   100% in    0s     2.47 MB/s  done
+```
+
+We wait a minute or so, we check the content of `input_bucket`, and we notice that it is empty.
+```
+$ s3cmd ls s3://input_bucket
+$
+```
+The pipeline has thus taken the image from `input_bucket` and processed it.
+
+We check if we have a trace of the image transformation in the database.
+On `terminal_C` we run the following command:
+```
+$ psql -h "$DB_HOST" -p "$DB_PORT" -U "$DB_USER" -d "$DB_NAME" -c "select * from log_records where negated_image_name like '%cat1%'"
```

Comment:
Same as above, cut the command in lines with `\` so it can be read better.

Reply:
Fixed in 7d67181791f56e3c02674275ab20fef7c6583629